File: app/src/main/java/com/vmloft/develop/app/match/glide/BlurTransformation.java
package com.vmloft.develop.app.match.glide;

import android.graphics.Bitmap;
import android.support.annotation.NonNull;

import com.bumptech.glide.load.engine.bitmap_recycle.BitmapPool;
import com.bumptech.glide.load.resource.bitmap.BitmapTransformation;

import com.vmloft.develop.library.tools.utils.bitmap.VMBlur;
import java.security.MessageDigest;

/**
 * Create by lzan13 on 2019/5/23 17:56
 *
 * 使用 RenderScript 模糊图片
 */
public class BlurTransformation extends BitmapTransformation {

    private static final String ID = "com.vmloft.develop.app.match.glide.BlurTransformation";
    private static final byte[] ID_BYTES = ID.getBytes(CHARSET);

    @Override
    protected Bitmap transform(BitmapPool pool, Bitmap toTransform, int outWidth, int outHeight) {
        Bitmap blurBitmap = VMBlur.stackBlurBitmap(toTransform, 20, 8, false);
        return blurBitmap;
    }

    @Override
    public void updateDiskCacheKey(@NonNull MessageDigest messageDigest) {
        messageDigest.update(ID_BYTES);
    }
}

File: vmcommon/src/main/java/com/vmloft/develop/library/common/image/BlurTransformation.kt
package com.vmloft.develop.library.common.image

import android.graphics.Bitmap

import com.bumptech.glide.load.Key
import com.bumptech.glide.load.engine.bitmap_recycle.BitmapPool
import com.bumptech.glide.load.resource.bitmap.BitmapTransformation

import com.vmloft.develop.library.common.BuildConfig
import com.vmloft.develop.library.tools.utils.bitmap.VMBlur.stackBlurBitmap

import java.security.MessageDigest

/**
 * Create by lzan13 on 2020/04/06 17:56
 * 描述：使用 RenderScript 模糊图片
 */
class BlurTransformation : BitmapTransformation() {
    private val id = "${BuildConfig.LIBRARY_PACKAGE_NAME}.glide.BlurTransformation"

    override fun transform(pool: BitmapPool, toTransform: Bitmap, outWidth: Int, outHeight: Int): Bitmap {
        val blurBitmap = stackBlurBitmap(toTransform, 15, 10, false)
        return blurBitmap!!
    }

    override fun updateDiskCacheKey(messageDigest: MessageDigest) {
        messageDigest.update(id.toByteArray(Key.CHARSET))
    }
}